File: php/src/Hasmap/GroupAnagrams.php
<?php


namespace IbrahimMH\LeetCode\Solution\Hasmap;

class GroupAnagrams{


    public function getList(array $words){

       $hashmap = [];

       foreach($words as $word){

            $letters = str_split($word);

            sort($letters);

            $key = implode('',$letters);

            if(!isset($hashmap[$key])){
            
                $hashmap[$key] = [];
            
            }

            $hashmap[$key][] = $word; 
       }

       return array_values($hashmap);
    
    }

}

File: python/hashmap/group_anagrams.py



def group_anagram(words: list) -> list:
    hashmap = {}

    for word in words:

        sortedWord = ''.join(sorted(word))
        
        if sortedWord in hashmap:

            hashmap[sortedWord].append(word)

        else:
            
            hashmap[sortedWord] = [word]
    
    return list(hashmap.values())
